frappe/tests/test_db_query.py

```diff
@@ -332,6 +333,16 @@ def test_is_set_is_not_set(self):
 		self.assertTrue({'name': 'Prepared Report'} in res)
 		self.assertFalse({'name': 'Property Setter'} in res)
 
+	def test_set_field_tables(self):
+		# Tests _in_standard_sql_methods method in test_set_field_tables
+		# The following query will break if the above method is broken
+		data = frappe.db.get_list("Web Form",
+			filters=[['Web Form Field', 'reqd', '=', 1]],
+			group_by='amount_field',
+			fields=['count(*) as count `amount_field` as name'],
+			order_by='count desc',
+			limit=50,
+		)
```

Comment:
Check if you are getting the expected result for this particular type of query.

Reply:
If the `_in_standard_sql_methods` method in `db_query.set_field_tables` does not work, this query will throw an exception... Otherwise the data sanity is maintained by the fact the `get_list` works as expected. If that is broken this will be broken too.

Reply:
Got the point that the query would fail if the implementation is wrong... but since there is no test that tests the output of the query with function in the field and has a child table in the filters, it is better to have the check of the output here.

Reply:
Perhaps I can add tests for get_list instead of checking this here?